conans/test/integration/package_id/test_validate.py

```diff
@@ -39,6 +40,59 @@ def validate(self):
         self.assertEqual(myjson[0]["binary"], BINARY_INVALID)
         self.assertEqual(myjson[0]["id"], 'INVALID')
 
+    def test_validate_header_only(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+            from conan.errors import ConanInvalidConfiguration
+            from conan.tools.build import check_min_cppstd
+            class Pkg(ConanFile):
+                settings = "os", "compiler"
+                options = {"shared": [True, False], "header_only": [True, False],}
+                default_options = {"shared": False, "header_only": True}
+
+                def package_id(self):
+                   if self.info.options.header_only:
+                       self.info.clear()
+
+                def validate(self):
+                    if self.info.options.get_safe("header_only") == "False":
+                        if self.info.settings.get_safe("os") == "Linux":
+                          raise ConanInvalidConfiguration("This package cannot exist in Linux")
+                        check_min_cppstd(self, 11)
+                        # These configurations are impossible
+                        if self.info.settings.os != "Windows" and self.info.options.shared:
+                            raise ConanInvalidConfiguration("shared is only supported under windows")
+
+                    # HOW CAN WE VALIDATE CPPSTD > 11 WHEN HEADER ONLY?
```

Comment:
This is an old question and I don't know if we have a new answer for it.

Reply:
Isn't it possible to check the ``settings`` directly here? Is it the ``check_min_cppstd()`` that checks ``self.info`` or is there other issue?

Reply:
`check_min_cppstd()` is using setting in develop but info in develop2.
About using settings here, is it viable/convenient with the compatible.py packages mechanism?